### PR TITLE
Differentiate Deep Dive workshops on map

### PR DESCRIPTION
**Pull request by islemaster**
[PLC-417](https://codedotorg.atlassian.net/browse/PLC-417) Displays different map info and pin colors depending on whether a CSF workshop is Intro or Deep Dive on code.org/professional-development-workshops/update.

Adds Deep Dive workshops to the map on this page.

Deep Dive workshops get a star on their pin.  We collapse workshops at the same location into one pin - in these cases if any of the workshops is deep dive, the pin gets the star.  Also adds the workshop subject to the tooltip.

Before:
image
image

After:
image
image

### CODE REVIEW

File: apps/src/sites/code.org/pages/views/workshop_search.js [hunk 1]
```diff
@@ -85,6 +91,13 @@ function createNewMarker(latLng, title, infoWindowContent) {
   return marker;
 }
 
+function iconForSubject(subject) {
+  if (subject === 'Deep Dive') {
```
Comment (breville) [inline apps/src/sites/code.org/pages/views/workshop_search.js]: Can we hoist `'Deep dive'` into a common string?

File: dashboard/app/controllers/api/v1/pd/workshops_controller.rb [hunk 2]
```diff
@@ -118,7 +118,9 @@ def k5_public_map_index
     # will be to show workshops from all subjects when deep-dive isn't specified.
     # But until then, when deep-dive isn't specified, we only show 'Intro'
     # workshops.
-    conditions[:subject] = params['deep_dive_only'] ? 'Deep Dive' : 'Intro'
+    if params['deep_dive_only']
```
Comment (breville) [inline dashboard/app/controllers/api/v1/pd/workshops_controller.rb:117]: Does this change the comment above it?

**Review by agealy — APPROVED** [review]

**Review by clareconstantine — APPROVED** [review]
I agree with brendan's comments, but otherwise this looks great!

File: dashboard/app/controllers/api/v1/pd/workshops_controller.rb (hunk 2 above)
Comment (islemaster) [reply dashboard/app/controllers/api/v1/pd/workshops_controller.rb:117]: Whoops, yes - it makes it irrelevant.  Removing!

**Review by breville — APPROVED** [review]

**Pull request merged**